src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.SyncNamespace
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.SyncNamespace), Shared]
+    internal sealed class CSharpSyncNamespaceCodeRefactoringProvider :
+        AbstractSyncNamespaceCodeRefactoringProvider<CSharpSyncNamespaceCodeRefactoringProvider, NamespaceDeclarationSyntax, CompilationUnitSyntax>
+    {
+        protected override ImmutableArray<ISymbol> GetDeclaredSymbolsInContainer(SemanticModel semanticModel, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var declarations = GetMemberDeclarationsInContainer(node);
+            var builder = ArrayBuilder<ISymbol>.GetInstance();
+            foreach (var declaration in declarations)
+            {
+                var symbol = semanticModel.GetDeclaredSymbol(declaration, cancellationToken);
+                builder.AddIfNotNull(symbol);
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        private IReadOnlyList<MemberDeclarationSyntax> GetMemberDeclarationsInContainer(SyntaxNode node)
+        {
+            if (node is NamespaceDeclarationSyntax namespaceDecl)
+            {
+                return namespaceDecl.Members;
+            }
+            else if (node is CompilationUnitSyntax compilationUnit)
+            {
+                return compilationUnit.Members;
+            }
+            else
+            {
+                return default;
```

Comment:
you return null here, but your caller doesn't check for that, so they can null ref.